test/unit/math/prim/scal/fun/locscale_transform_test.cpp

```diff
@@ -0,0 +1,52 @@
+#include <stan/math/prim/scal.hpp>
+#include <test/unit/util.hpp>
+#include <gtest/gtest.h>
+#include <limits>
+
+TEST(prob_transform, locscale) {
+  EXPECT_FLOAT_EQ(2.0 - 5.0 * 1.0,
+                  stan::math::locscale_constrain(-1.0, 2.0, 5.0));
+
+  EXPECT_FLOAT_EQ(1.7, stan::math::locscale_constrain(1.7, 0, 1));
+}
+TEST(prob_transform, locscale_j) {
+  double lp = -17.0;
+  double L = 2.0;
+  double U = 5.0;
+  double x = -1.0;
+  EXPECT_FLOAT_EQ(L + U * x, stan::math::locscale_constrain(x, L, U, lp));
+  EXPECT_FLOAT_EQ(-17.0 + log(U), lp);
+
+  double lp1 = -12.9;
+  EXPECT_FLOAT_EQ(1.7, stan::math::locscale_constrain(1.7, 0, 1, lp1));
+  EXPECT_FLOAT_EQ(-12.9, lp1);
+}
+TEST(ProbTransform, locscaleException) {
```

Comment:
add tests to check that inf mu triggers exception

Reply:
Also done.